src/test/java/com/joss/conductor/mobile/ArtifactUtilTest.java

```diff
@@ -0,0 +1,168 @@
+
+        Assertions.assertThat(testPart)
+                .isEqualTo(test);
+        try {
+            Assertions.assertThat(ArtifactUtil.SDF.parse(datePart))
+                    .isToday();
+        } catch (ParseException e) {
+            Assert.fail("Exception thrown:" + e.toString());
+        }
+    }
+
+    @Test
+    public void test_file_name_creation_is_unique() {
+        String path = "Name of the Test";
+        String testName = "Error message of the test";
+
+        String filepath1 = ArtifactUtil.artifactPathForTest(path, testName);
+        WaitUtil.wait(100, "Waiting " + 100 + " millis");
+        String filepath2 = ArtifactUtil.artifactPathForTest(path, testName);
+
+        Assertions.assertThat(filepath1)
+                .isNotEqualTo(filepath2);
+    }
+
+    @Test
+    public void test_file_name_less_than_100_characters() {
+        String path = "name_of_test_name_of_test_name_of_test_name_of_test_name_of_test_name_of_test_";
+
+        String filePath = ArtifactUtil.artifactPathForTest(path, "");
+
+        String retPath = Paths.get(filePath).getFileName().toString();
+        Assertions.assertThat(retPath.length())
+                .isEqualTo(100);
+    }
+
+    @Test
+    public void test_with_path_returns_path_filename_timestamp() {
+        String path = "path_for_this_artifact";
+        String testName = "name_of_test";
+
+        String filePath = ArtifactUtil.artifactPathForTest(path, testName, ".png");
+
+        Path retFilePath = Paths.get(filePath);
+        String retFile = retFilePath.getFileName().toString();
+        String[] fileParts = retFile.split("-");
+        String retPath = retFilePath.getName(retFilePath.getNameCount() - 2).toString();
+
+        Assertions.assertThat(fileParts[0])
+                .isEqualTo(testName);
+        try {
+            Assertions.assertThat(ArtifactUtil.SDF.parse(fileParts[1]))
+                    .isToday();
+        } catch (ParseException e) {
+            Assert.fail("Exception thrown:" + e.toString());
+        }
+        Assertions.assertThat(retPath)
+                .isEqualTo(path);
+    }
+
+    @Test
+    public void test_file_name_sanitizes_special_characters() {
+        String path = "test_with:_?*s<p>ec\"ial||_/cha\\racters";
+
+        String filePath = ArtifactUtil.artifactPathForTest(path, "");
+
+        String retPath = Paths.get(filePath).getFileName().toString();
+        retPath = retPath.substring(0, retPath.indexOf('-'));
+        Assertions.assertThat(retPath)
+                .isEqualTo("test_with+_++s+p+ec+ial++_+cha+racters");
+    }
+
+    @Test
+    public void test_file_name_and_path_creation_is_less_than_100_chars() {
+        String path = "name_of_test_name_of_test_name_of_test_name_of_test_name_of_test_name_of_test_longy_longy_longy_long_long_long";
+        String testName = "error_message_of_test_error_message_of_test_error_message_of_test_error_message_of_test_more_long_than_120_characters_why";
```

Comment:
Do we have any guide lines about maximum line length in our Java code? 100 characters is a pretty popular standard (https://google.github.io/styleguide/javaguide.html#s4.4-column-limit)

Reply:
We don't, other than to say they should be of reasonable length. 

Would be hard to write a unit test that needs to test truncation of a string to 100 characters in less than 100 characters though ;)